src/pymor/models/iosys.py

```diff
@@ -222,12 +222,8 @@ def __init__(self, A, B, C, D=None, E=None, cont_time=True,
                          estimator=estimator, visualizer=visualizer,
                          cache_region=cache_region, name=name)
 
-        self.A = A
-        self.B = B
-        self.C = C
-        self.D = D
-        self.E = E
-        self.solver_options = solver_options
+        self.D = D  # can't use autoassign since D might have been modified above
```

Comment:
How about replacing above `D = D or ZeroOperator(C.range, B.source)` with `if D:` and using `self.D = D or ZeroOperator(C.range, B.source)` here?